src/checkqueue.h

```diff
@@ -157,6 +161,16 @@ class CCheckQueue
             condWorker.notify_all();
     }
 
+    void Interrupt()
+    {
+        {
+            std::lock_guard<std::mutex> lock(mutex);
+            interrupted = true;
+        }
+        condMaster.notify_all();
```

Comment:
tiny nit: only one master thread so maybe `condMaster.notify_one()` is more consistent. No difference to code though.